Quake/cl_parse.c

```diff
@@ -1822,9 +1822,11 @@ void CL_ParseServerMessage (void)
 		case svc_updatefrags:
 			Sbar_Changed ();
 			i = MSG_ReadByte ();
-			if (i >= cl.maxclients)
-				Host_Error ("CL_ParseServerMessage: svc_updatefrags > MAX_SCOREBOARD");
-			cl.scores[i].frags = MSG_ReadShort ();
+			j = MSG_ReadShort ();
+			if (i < 0 || i >= cl.maxclients)
+				Con_DPrintf ("CL_ParseServerMessage: svc_updatefrags > MAX_SCOREBOARD\n");
```

Comment:
Why was this necessary? Do we hit this error in the r-release version?

Reply:
Was hitting this when loading mg1/id2 savegames (receiving a `svc_updatefrags` when `cl.maxclients` was still 0), but it only happens when using the fte666/fte999 protocols, so QS doesn't need it. Strangely enough, setting `cl_nopext 1` doesn't fix it.

Reply:
I tried running mg1 with the minimal three changes in qs, saved and loaded,
got no errors: is the error random or guaranteed to happen?

Reply:
In vkQuake it happens every time, but I haven't seen it at all in QS, so it's probably safe to remove this change from the QS patch.

Reply:
BTW, applied the same three changes in here to qs too.

Reply:
> In vkQuake it happens every time, but I haven't seen it at all in QS, so it's probably safe to remove this change from the QS patch.

OK then.

Reply:
> In vkQuake it happens every time, but I haven't seen it at all in QS, 

I actually wonder why it is happening in vkquake - that you are saying it still happens with `cl_nopext 1`, curious

Reply:
I've looked into this a bit more, and it seems that when the FTE protocol extensions are active, `SV_SendServerinfo` will first ask the client what extensions it supports
https://github.com/Novum/vkQuake/blob/18df3c1cdcfea8da4a33750cd781ab9a30eed1f7/Quake/sv_main.c#L1289-L1295
and only then send the actual server info message (`SV_Pext_f`)
https://github.com/Novum/vkQuake/blob/18df3c1cdcfea8da4a33750cd781ab9a30eed1f7/Quake/sv_main.c#L1512-L1513
The problem is that on the same frame that the client connects and the extension query is sent, the server will also send `svc_updatefrags` messages (from `SV_UpdateToReliableMessages`, called from `SV_SendClientMessages`)
https://github.com/Novum/vkQuake/blob/18df3c1cdcfea8da4a33750cd781ab9a30eed1f7/Quake/sv_main.c#L2263-L2279
The `svc_updatefrags` arrives at the client before `svc_serverinfo`, which means that `cl.maxclients` is still 0, hence the error.

A possible solution (other than turning the error into a warning) might be to change
https://github.com/Novum/vkQuake/blob/18df3c1cdcfea8da4a33750cd781ab9a30eed1f7/Quake/sv_main.c#L2270
to
```c
if (!client->active || client->sendsignon == PRESPAWN_FLUSH)
```
to avoid sending `svc_updatefrags` before `svc_serverinfo`. It seems to work, but I've only done some limited testing and I'm not exactly sure of the implications, especially for multiplayer mode. I'd say that changing the error to a warning still seems like the safest solution, but I'm also wondering why this isn't triggered with any other mod.

Reply:
@Shpoike: comments?

Reply:
I tried QSS, and it doesn't crash when loading. The diffference is that it's [checking](https://github.com/Shpoike/Quakespasm/blob/c48936385a4428ce701059a11215780ec3cdcc00/quakespasm/Quake/sv_main.c#L2916) `client->knowntoqc` instead of `client->active` - I'll commit this and rebase.

Reply:
OK